fusequery/query/src/sql/plan_parser.rs

```diff
@@ -608,9 +611,10 @@ impl PlanParser {
                     })
                 };
 
+                // TODO: Move ReadSourcePlan to SelectInterpreter
                 scan.and_then(|scan| match scan {
                     PlanNode::Scan(ref scan) => table
-                        .read_plan(self.ctx.clone(), scan)
+                        .read_plan(self.ctx.clone(), scan, self.ctx.get_max_threads()? as usize)
```

Comment:
We already got ctx, why need to pass `max_thread` in ?

Reply:
In standalone mode, the maximum concurrency is `max_threads`. 
For cluster mode, the maximum concurrent number should be `max_threads * nodes_size`. 
We will re plan the plan in the plan_scheduler.

``` rust
let table = ctx.get_table(&plan.db, &plan.table)?;
if !table.is_local() {
    let new_partitions_size = ctx.get_max_threads()? as usize * cluster_nodes.len();
    let new_read_source_plan = table.read_plan(ctx.clone(), &*plan.scan_plan, new_partitions_size)?; 
    // We always put adjacent partitions in the same node
    let new_partitions = &new_read_source_plan.partitions;
    let mut nodes_partitions = HashMap::new();
    let partitions_pre_node = new_partitions.len() / cluster_nodes.len();
```

reference: https://github.com/datafuselabs/datafuse/pull/469/files#diff-d9bf8b037f42cfc24f1b4e8138e966f80aae1b68deb6fe82ceb0d2ad7a93a806R236